.gitignore

```diff
@@ -2,9 +2,11 @@ build/
 local-etc/
 /agent
 internal/monitors/**/template.go
+internal/monitors/**/generatedMetadata.go
```

Comment:
I'm not so sure the generated files should be ignored.  If we put them in the git repo, there should definitely be a CI check to ensure they are up to date, but it would be kind of nice to have them.  Same thing with the collectd template.go files, not sure why I ignored them to begin with.

Reply:
Here is a Reddit thread discussing it: https://www.reddit.com/r/golang/comments/8mbi47/do_you_commit_all_the_generated_code_in_your/

Reply:
I personally consider them build artifacts and not something to be committed (unless it's a library that something else is going to depend on).  I don't really care either way though as long as it's consistent with however the collectd ones are done.

Reply:
Ok, we can leave them out and it will be easy to add them in later if we find some convenient reason to do so.  I think in CI we can generate them fairly easily, but if it greatly complicates CI jobs to omit them from the repo, that would be my main reason to include them.